src/lava/magma/runtime/_c_message_infrastructure/csrc/channel/dds/dds_channel.cc

```diff
@@ -9,6 +9,23 @@
 
 namespace message_infrastructure {
 
+DDSChannel::DDSChannel(const std::string &src_name,
+                       const std::string &dst_name,
+                       const size_t &size,
+                       const size_t &nbytes,
+                       const DDSTransportType &dds_transfer_type,
+                       const DDSBackendType &dds_backend) {
+  LAVA_DEBUG(LOG_DDS, "Creating DDSChannel...\n");
+
+  dds_ = GetDDSManagerSingleton().AllocDDS(
+                                  "dds_topic_" + std::to_string(std::rand()),
```

Comment:
put invoking `std::rand()` under `DDSManager` that is same behavior with `SharedMemManager`.

Reply:
Because one of the two constructors of DDSchannel will use the topic name entered by the user. the std::rand() can't be put in DDSManager unless add a if statement in AllocDDS.